app/Http/Controllers/TaskController.php

```diff
@@ -3,13 +3,117 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use App\Task;
+use Illuminate\Support\Facades\App;
 
 class TaskController extends Controller
 {
+    //getTaskView method
+    public function getTaskView () 
+    {
+        return view('tasks');
+    }
+
     //postTask method
-    public function postTask ()
+    public function postTask (Request $request)
+    {   
+        $request->validate(
+            [
+                'name' => 'required|min:2|max:50',
+            ]
+        );
+
+        $task = new Task;
+        $task->name = $request->name;
+        $task->save();
+
+        $locale = App::getLocale();
+
+        if ($locale == 'en')
+        {
+            return redirect('/')->with('notice', 'Task successfully created');
+        }
+        elseif ($locale == 'vi') 
+        {
+            return redirect('/')->with('notice', 'Đã tạo công việc mới');
+        }
+    }
+
+    public function getAllTasks ()
     {
+        $tasks = Task::all();
 
+        return view('tasks', ['tasks' => $tasks]);
     }
-}
 
+    public function getUpdateTask ($id)
+    {
+        $taskById = Task::find($id);
+
+        if ($taskById == null)
+        {
+            $errorNotify = "Can't find this task";
+            return view('error-landing', ['error' => $errorNotify]);
+        }
+        else
+        {
+            return view('update', ['task' => $taskById]);
+        }
+    }
+
+    public function postUpdateTask (Request $request, $id)
+    {
+        $request->validate(
+            [
+                'name' => 'required|min:2|max:50',
+            ]
+        );
+
+        $task = Task::find($id);
+        $task->name = $request->name;
+        $task->save();
+
+        $locale = App::getLocale();
+
+        if ($locale == 'en')
+        {
+            return redirect('/')->with('notice', 'Task successfully updated');
+        }
+        elseif ($locale == 'vi') 
+        {
+            return redirect('/')->with('notice', 'Đã cập nhật công việc');
+        }
+    }
+
+    public function getDeleteTask ($id)
+    {
+        $taskById = Task::find($id);
+
+        if ($taskById == null)
+        {
+            $errorNotify = "Can't find this task";
+            return view('error-landing', ['error' => $errorNotify]);
+        }
+        else
+        {
+            return view('delete', ['task' => $taskById]);
+        }
+    }
+
+    public function postDeleteTask ($id)
+    {
+        $task = Task::find($id);
+        $task->delete();
+
+        $locale = App::getLocale();
```

Comment:
Thiết lập Locale cho toàn app luôn chứ không check cho từng action như này e

Reply:
Dạ anh